### DISCUSSION TITLE
Welcome to kitt4sme Discussions!

### DISCUSSION THREAD

**Discussion by c0c0n3** [c0c0n3/kitt4sme#12] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

**Comment by etoschi** [comment]
@[c0c0n3](https://github.com/c0c0n3)
We have other questions.
Thank you.
ArgoCD Configuration
 

I followed the steps in the bootstrap.md guide but ArgoCD doesn’t seem to refresh contents, or in any case show connection to our corporate gitlab repository.

From ArgoCD UI I see that there’s still reference to the [github.com/c0c0n3/kitt4sme.live](http://github.com/c0c0n3/kitt4sme.live) repository, though I changed the values in deployment/mesh-infra/_replacements_/custom-urls.yaml as suggested by the guide.

 

![image001](https://user-images.githubusercontent.com/101259854/166196672-d535e398-b330-4dee-98fc-447e4c05eea3.png)

Graphical user interface, application

Description automatically generated

 

How do I check if ArgoCD is correctly configured? I guess that I could “manually” setup connection to the git repository through the ArgoCD UI but I wonder why these manual config steps are not mentioned in the bootstrap guide and for this reason I guess this is not the correct procedure.

I also noticed that while configuring the repository URL in custom-urls.yaml, there’s no mention to configuring credentials to access this repository (username and password or ssh key), am I missing something?

From the UI, if I try to define a new connection via Repositories > CONNECT REPO USING SSH/HTTPS, what is the right value to select for “Project”?

 
![image002](https://user-images.githubusercontent.com/101259854/166196746-813927de-1f8e-4cb4-83af-6c094ffa723b.png)


Graphical user interface, text, application

Description automatically generated

 

SealedSercretController image pull issue
 

I noticed the following error ImagePullBackOff when examining the just installed demo cluster :

 

kubectl get pods -n kube-system

NAME                                         READY   STATUS             RESTARTS   AGE

hostpath-provisioner-566686b959-h5c2k        1/1     Running            1          25h

calico-kube-controllers-f7868dd95-rj8jv      1/1     Running            1          25h

coredns-7f9c69c78c-xjb52                     1/1     Running            1          25h

calico-node-5ckf2                            1/1     Running            1          25h

reloader-55885ccf44-5xmjl                    1/1     Running            0          25h

sealed-secrets-controller-75445c88df-f4249   0/1     ImagePullBackOff   0          25h

 

kubectl describe pod sealed-secrets-controller-75445c88df-f4249 -n kube-system

 

...

Events:

  Type    Reason   Age                    From     Message

  ----    ------   ----                   ----     -------

  Normal  Pulling  52m (x227 over 19h)    kubelet  Pulling image "[quay.io/bitnami/sealed-secrets-controller:v0.17.3](http://quay.io/bitnami/sealed-secrets-controller:v0.17.3)"

  Normal  BackOff  2m6s (x5246 over 19h)  kubelet  Back-off pulling image "[quay.io/bitnami/sealed-secrets-controller:v0.17.3](http://quay.io/bitnami/sealed-secrets-controller:v0.17.3)"

**Comment by c0c0n3** [comment]
hi @etoschi!

Thanks alot for your questions and reporting issues. I've split your comments into two questions and opened an issue (bug report) too:

* [ArgoCD custom repo URL not working](https://github.com/c0c0n3/kitt4sme.live/issues/145)
* [ArgoCD manual configuration](https://github.com/c0c0n3/kitt4sme.live/discussions/146)
* [Sealed Secrets image pull issues](https://github.com/c0c0n3/kitt4sme.live/discussions/144)

If you could please have a look there and carry on the discussion over there if needed. Thanks a million!

**Reply by etoschi** [reply]
@the-mi
Sealed Secrets Pull Issue [github #144](https://github.com/c0c0n3/kitt4sme.live/discussions/144)

 

I tried to delete the pod as suggested, but still getting ImagePullBackOff – there’s a difference, though, in that now the Events: section in the describe command for the pod provides more output, in particular showing a 401 UNAUTHORIZED message

 

Events:

  Type     Reason     Age                From               Message

  ----     ------     ----               ----               -------

  Normal   Scheduled  60s                default-scheduler  Successfully assigned kube-system/sealed-secrets-controller-75445c88df-8sq7m to algreenwaste

  Normal   Pulling    15s (x3 over 59s)  kubelet            Pulling image "[quay.io/bitnami/sealed-secrets-controller:v0.17.3](http://quay.io/bitnami/sealed-secrets-controller:v0.17.3)"

  Warning  Failed     13s (x3 over 58s)  kubelet            Failed to pull image "[quay.io/bitnami/sealed-secrets-controller:v0.17.3](http://quay.io/bitnami/sealed-secrets-controller:v0.17.3)": rpc error: code = Unknown desc = failed to pull and unpack image "[quay.io/bitnami/sealed-secrets-controller:v0.17.3](http://quay.io/bitnami/sealed-secrets-controller:v0.17.3)": failed to resolve reference "[quay.io/bitnami/sealed-secrets-controller:v0.17.3](http://quay.io/bitnami/sealed-secrets-controller:v0.17.3)": unexpected status code [manifests v0.17.3]: 401 UNAUTHORIZED

  Warning  Failed     13s (x3 over 58s)  kubelet            Error: ErrImagePull

  Normal   BackOff    1s (x3 over 58s)   kubelet            Back-off pulling image "[quay.io/bitnami/sealed-secrets-controller:v0.17.3](http://quay.io/bitnami/sealed-secrets-controller:v0.17.3)"

  Warning  Failed     1s (x3 over 58s)   kubelet            Error: ImagePullBackOff

 

I tried several times to delete the pod, just to be sure, but always getting the same result.

 

ArgoCD custom repo URL not working [#145](https://github.com/c0c0n3/kitt4sme.live/issues/145)

 

Unfortunately I can’t give you access to the copy on the corporate gitlab repository, since this is not a public repo.In an earlier attempt to install the cluster I forked the repo on github at https://github.com/cmelchiorre/kitt4sme.live which is a public repo, but this was just an early attempt, so I cannot guarantee that’s exactly the same, beyond having followed the same steps in te tutorial bootstrap.md (except, of course, the exact git URL and IP address for the cluster I set in custom-config which are different from the current git/cluster configuration)

concerning the sanity check: I’m pretty sure I did those in the order you mention; My doubt are more than I had to change some other command in the bootstrap guide (for instance replacing the URL in the nix tools installation command echo 'nix shell github:c0c0n3/kitt4sme.live?dir=nix' > ~/tools.sh (which I didn’t); might this be the problem? (I’m not familiar with Nix and didn’t really understand this command at the moment, so I just copy-and-pasted, now I realize that this could be the issue). If it’s so, will I solve the problem if I just change the url in the above nix shell command repeat the following steps in the guide?

 

ArgoCD manual configuration [#146](https://github.com/c0c0n3/kitt4sme.live/discussions/146)

 

Ok, but I have a doubt it’s taking the configuration from the wrong place: I mean from [github.com/c0c0n3/kitt4sme.live](http://github.com/c0c0n3/kitt4sme.live) instead of the url I changed in custom-urls.yaml; this poses a “chicken and egg” like problem, since (a) I must make changes yamls on my private repository to change configuration for ArgoCD and  (b) until ArgoCD isn’t correctly configured, it won’t take changes I make to configuration in yamls. I wonder If I followed some steps in the guide wrongly, and have to specify the corporate git repository URL somewhere else beyond the custom-urls.yaml

**Comment by c0c0n3** [comment]
Hi @etoschi, thanks for your comments, much appreciated. Like I said earlier, I broke down this discussion in 3 separate pieces:

* [ArgoCD custom repo URL not working](https://github.com/c0c0n3/kitt4sme.live/issues/145)
* [ArgoCD manual configuration](https://github.com/c0c0n3/kitt4sme.live/discussions/146)
* [Sealed Secrets image pull issues](https://github.com/c0c0n3/kitt4sme.live/discussions/144)

If you could please have a look there and carry on the discussion over there if needed. Thanks a million!
